includes/class-wc-gateway-komoju-ipn-handler.php

```diff
@@ -113,115 +131,91 @@ protected function validate_amount( $order, $amount ) {
+		update_post_meta( $order->get_id(), sprintf( __( 'Payment %s via IPN.', 'komoju-woocommerce' ), wc_clean( webhookResponse.status() ) ) );
 	}
 
 	/**
 	 * Handle a refunded order
 	 * @param  WC_Order $order
+	 * @param  WC_Gateway_Komoju_Webhook_Event $webhookEvent Webhook event data
 	 */
-	protected function payment_status_refunded( $order, $posted ) {
+	protected function payment_status_refunded( $order, $webhookEvent ) {
 		// Only handle full refunds, not partial
-		WC_Gateway_Komoju::log( 'Only handling full refund. Controlling that order total equals amount refunded. Does '.$order->get_total().' equals '.$posted['grand_total'].' ?' );
-		if ( $order->get_total() == ( $posted['grand_total'] ) ) {
+		WC_Gateway_Komoju::log( 'Only handling full refund. Controlling that order total equals amount refunded. Does '.$order->get_total().' equals '.$webhookEvent->grand_total().' ?' );
+		if ( $order->get_total() == ( $webhookEvent->amount_refunded() ) ) {
 
 			// Mark order as refunded
-			$order->update_status( 'refunded', sprintf( __( 'Payment %s via IPN.', 'komoju-woocommerce' ), strtolower( $posted['status'] ) ) );
-
-			/*$this->send_ipn_email_notification(
-				sprintf( __( 'Payment for order #%s refunded/reversed', 'woocommerce' ), $order->get_order_number() ),
-				sprintf( __( 'Order #%s has been marked as refunded - Komoju reason code: %s', 'woocommerce' ), $order->get_order_number(), $posted['reason_code'] )
-			);*/
+			$order->update_status( 'refunded', sprintf( __( 'Payment %s via IPN.', 'komoju-woocommerce' ), strtolower( $webhookEvent->status() ) ) );
 		}
 	}
 
 	/**
 	 * Save important data from the IPN to the order
 	 * @param WC_Order $order
+	 * @param  WC_Gateway_Komoju_Webhook_Event $webhookEvent Webhook event data
 	 */
-	protected function save_komoju_meta_data( $order, $posted ) {
-		if ( ! empty( $posted['tax'] ) ) {
-			update_post_meta( $order->id, 'Tax', wc_clean( $posted['tax'] ) );
+	protected function save_komoju_meta_data( $order, $webhookEvent ) {
+		if ( ! empty( $webhookEvent->tax() ) ) {
+			update_post_meta( $order->get_id(), 'Tax', wc_clean( $webhookEvent->tax() ) );
 		}
-		if ( ! empty( $posted['amount'] ) ) {
-			update_post_meta( $order->id, 'Amount', wc_clean( $posted['amount'] ) );
+		if ( ! empty( $webhookEvent->amount() ) ) {
+			update_post_meta( $order->get_id(), 'Amount', wc_clean( $webhookEvent->amount() ) );
 		}
-		if ( ! empty( $posted['additional_information'] ) ) {
-			update_post_meta( $order->id, 'Additional info', wc_clean( print_r( $posted['additional_information'], true) ) );
+		if ( ! empty( $webhookEvent->additional_information() ) ) {
+			update_post_meta( $order->get_id(), 'Additional info', wc_clean( print_r( $webhookEvent->additional_information(), true) ) );
 		}
 	}
-
-	/**
-	 * Send a notification to the user handling orders.
-	 * @param  string $subject
-	 * @param  string $message
-	 */
-	/*protected function send_ipn_email_notification( $subject, $message ) {

```

Comment:
I deleted this code as it's been around for 3 years and wasn't being used, so I figured it probably wasn't going to be used any time soon.